wgpu/src/lib.rs

```diff
@@ -3373,6 +3373,17 @@ impl Queue {
         Context::queue_write_texture(&*self.context, &self.id, texture, data, data_layout, size)
     }
 
+    /// Schedule a copy of data from `image` into `texture`
+    #[cfg(all(target_arch = "wasm32", not(feature = "webgl")))]
```

Comment:
Currently we don't support choosing between WebGPU and WebGL at runtime, so this code is right given our context.

Reply:
Good to know!

Reply:
Removing a function violates the rules for features:

> … features should be _additive._ That is, enabling a feature should not disable functionality, and it should usually be safe to enable any combination of features. A feature should not introduce a [SemVer-incompatible change](https://doc.rust-lang.org/cargo/reference/features.html#semver-compatibility).

— https://doc.rust-lang.org/cargo/reference/features.html

But, can't this function be implemented on WebGL? [MDN says `ImageBitmap` is a possible texture data source](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D).

Reply:
It can sort of be implemented - WebGPU supports copying from more sources than WebGL does - I agree we should implement it on WebGL but violating the rule of features until we implement it is okay. I don't want to force them to dig deep into wgc/hal to be able to implement this feature on WebGPU.